apps/keyboard/js/layouts/fa.js

```diff
@@ -0,0 +1,98 @@
+Keyboards.fa = {
+  label: 'Persian',
+  shortLabel: 'فا',
+  menuLabel: 'فارسی',
+  secondLayout: true,
+  specificCssRule: true,
+  types: ['text', 'url', 'email'],
+  alternateLayoutKey: '۱۲۳',
+  basicLayoutKey: 'اب‌پ',
+  width: 11,
+  lang: 'fa',
+  needsCommaKey: true,
+  alt : {
+    'ض': ['۱', '1'],
+    'ص': ['۲', '2'],
+    'ث': ['۳', '3'],
+    'ق': ['۴', '4'],
+    'ف': ['۵', '5'],
+    'غ': ['۶', '6'],
+    'ع': ['۷', '7'],
+    'ه': ['ۀ', 'ة', '۸', '8'],
+    'خ': ['۹', '9'],
+    'ح': ['۰', '0'],
+    'ا': ['آ', 'أ', 'ء'],
+    'ی': ['ئ','ي'],
+    'ت': 'ة',
+    'ک': 'ك',
+    '.': 'ًٌٍَُِّْءٔٓ',
+    '،': ':!؟؛-«»'
+  },
+  textLayoutOverwrite: {
+    ',': '،'
+  },
+  keys: [
+    [
+      { value: 'ض' }, { value: 'ص' }, { value: 'ث' }, { value: 'ق' },
+      { value: 'ف' }, { value: 'غ' }, { value: 'ع' }, { value: 'ه' },
+      { value: 'خ' }, { value: 'ح' }, { value: 'ج' }
+    ], [
+      { value: 'ش' }, { value: 'س' }, { value: 'ی' }, { value: 'ب' },
+      { value: 'ل' }, { value: 'ا' }, { value: 'ت' }, { value: 'ن' },
+      { value: 'م' }, { value: 'ک' }, { value: 'گ' }
+    ], [
+      { value: 'ظ' }, { value: 'ط' }, { value: 'ژ' }, { value: 'ز' },
+      { value: 'ر' }, { value: 'ذ' }, { value: 'د' }, { value: 'پ' },
+      { value: 'و' }, { value: 'چ' }, { value: '⌫', ratio: 1, keyCode: KeyEvent.DOM_VK_BACK_SPACE }
+    ], [
+      { value: '&nbsp', ratio: 8, keyCode: KeyboardEvent.DOM_VK_SPACE },
+      { value: '\u25C2\u2507\u25B8', ratio: 1, keyCode: '8204', ariaLabel: 'نیم‌فاصله' },
+      { value: '↵', ratio: 2, keyCode: KeyEvent.DOM_VK_RETURN }
+    ]
+  ],
+  pages: [undefined, {
```

Comment:
Since you are defining a second page, it make sense for you to define the first page here (instead of putting a "undefined" and have LayoutNormalizer to move the properties of the layouts here).

FYI https://github.com/mozilla-b2g/gaia/blob/master/apps/keyboard/js/keyboard/layout_normalizer.js#L218-L220